express-notes-api/index.js

```diff
@@ -0,0 +1,120 @@
+const express = require('express');
+const data = require('./data.json');
+const app = express();
+const fs = require('fs');
+
+const JSONMiddleware = express.json();
+app.use(JSONMiddleware);
+
+app.get('/api/notes', (req, res) => {
+  const array = [];
+  for (const prop in data.notes) {
+    array.push(data.notes[prop]);
+  }
+  res.json(array);
+});
+
+app.get('/api/notes/:id', (req, res) => {
+  const id = parseFloat(req.params.id);
+  if (!Number.isInteger(id) || id < 1) {
+    res.status(400).json({
+      error: 'id must be a positive integer'
+    });
+  } else {
+    const entry = data.notes[id];
+    if (entry === undefined) {
+      res.status(404).json({
+        error: `cannot find note with id ${id}`
+      });
+    }
+    res.status(200).json(entry);
```

Comment:
Be sure to include `else` or `return` appropriately so that it's impossible for the server to attempt to respond more than once to a given request.